src/sentry/issues/priority.py

```diff
@@ -93,32 +97,28 @@ def get_priority_for_ongoing_group(group: Group) -> PriorityLevel | None:
     if not features.has("projects:issue-priority", group.project, actor=None):
         return None
 
-    previous_priority_history = (
-        GroupHistory.objects.filter(
-            group_id=group.id, status__in=PRIORITY_TO_GROUP_HISTORY_STATUS.values()
-        )
-        .order_by("-date_added")
-        .first()
-    )
+    previous_priority_history = GroupHistory.objects.filter(
+        group_id=group.id, status__in=PRIORITY_TO_GROUP_HISTORY_STATUS.values()
+    ).order_by("-date_added")
 
-    new_priority = (
-        [
-            priority
-            for priority, status in PRIORITY_TO_GROUP_HISTORY_STATUS.items()
-            if status == previous_priority_history.status
-        ][0]
-        if previous_priority_history
-        else get_initial_priority(group)
-    )
+    new_priority = None
+    if len(previous_priority_history) > 1:
```

Comment:
Looking at django docs I think `len()` will query the entire set. If you do `.count()` it will only query the count. You could also just do `previous_priority_history[1]` and catch a raised error if it doesn't exist which should only query that single item